marklogic-client-api/src/test/java/com/marklogic/client/test/MarkLogicVersion.java

```diff
@@ -0,0 +1,52 @@
+package com.marklogic.client.test;
```

Comment:
This is a variation of the DHF class with the same name that DHF uses to conditionally run tests. Cleaned it up a bit and added a bunch of tests to verify that it's able to parse different versions correctly.